src/platform/plugins/shared/discover/public/context_awareness/profile_providers/common/patterns/profile.ts

```diff
@@ -56,18 +54,26 @@ export const createPatternDataSourceProfileProvider = (
         {
           id: 'patterns-action-view-docs-in-discover',
           getDisplayName: () =>
-            i18n.translate('discover.docViews.patterns.cellAction.viewDocsInDiscover', {
-              defaultMessage: 'View docs in Discover',
+            i18n.translate('discover.docViews.patterns.cellAction.viewResults', {
+              defaultMessage: 'View matching results',
             }),
           getIconType: () => 'discoverApp',
+          isCompatible: (context) => {
+            const { query, field } = context;
+            if (!isOfAggregateQueryType(query) || field === undefined) {
+              return false;
+            }
+            const patternColumns = getCategorizeColumns(query.esql);
+            return patternColumns.includes(field.name);
+          },
```

Comment:
Similar to `getCellRenderers`, you can access `patternColumns` from the second param of `getAdditionalCellActions` directly instead of re-parsing the query:
```ts
getAdditionalCellActions: (prev, { context: { patternColumns } }) => () => {
...
isCompatible: (context) => {
  const { field } = context;
  return patternColumns.includes(field.name);
},
```